Description: munpack should return 1 on errors.
Author: Fabio Augusto De Muzio Tobich <ftobich@gmail.com>
Bug-Debian: https://bugs.debian.org/527555
Last-Update: 2020-05-22
--- mpack-1.6.orig/unixunpk.c
+++ mpack-1.6/unixunpk.c
@@ -90,6 +90,7 @@ int main(int argc, char **argv)
 	file = fopen(argv[optind], "r");
 	if (!file) {
 	    perror(argv[optind]);
+		exit(1);
 	}
 	else {
 	    didchat = 0;
